Python/signal_generation/test/test_signal_generation.py

```diff
@@ -0,0 +1,183 @@
+""" Provides tests to demonstrate usage of modules inside package
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+
+import signal_generation.conversion as sg_conv
+import signal_generation.noise as sg_noise
+import signal_generation.positioning as sg_pos
+import signal_generation.receiver as sg_rec
+import signal_generation.source as sg_src
+
+
+@pytest.mark.plot
+def test_given_multiple_sources_then_multiple_frequencies():
+    positions = np.array(
+        [
+            sg_pos.Position(),
+        ]
+    )
+
+    hydro_array = sg_rec.HydrophoneArray(
+        positions=positions,
+    )
+
+    pinger_position_one = sg_pos.Position(
+        x=3.0,
+        y=-2.2,
+        z=1.2,
+    )
+    pinger_position_two = sg_pos.Position(
+        x=3.0,
+        y=-2.2,
+        z=4.2,
+    )
+    pinger = [
+        sg_src.Pinger(
+            frequency=10.0,  # [kHz]
+            period=30.0,
+            pulse_length=15.0,
+            use_window=True,
+            position=pinger_position_one,
+        ),
+        sg_src.Pinger(
+            frequency=20.0,  # [kHz]
+            period=30.0,
+            pulse_length=15.0,
+            use_window=True,
+            position=pinger_position_two,
+        ),
+    ]
+
+    result = sg_rec.generate_signals(
+        sources=pinger,
+        receivers=hydro_array,
+        output_length=2000,
+        sound_speed=1500.0,
+    )
+
+    fig, axs = plt.subplots(2, 1)
+    fig.tight_layout()
+    axs[0].plot(result[0])
+    axs[0].set_xlabel("n")
+    axs[0].set_xlim([250, 500])
+    axs[1].plot(abs(np.fft.fft(result[0])))
+    axs[1].set_xlabel("frequency bins")
```

Comment:
Basically, I was just trying to visualize what it would like in time versus frequency domain. For many of these tests I didn't have an easy way to test the behavior/functionality without visual and manual confirmation. Then I just plotted some segments of the signal.